CONTRIBUTING.md

```diff
@@ -1,153 +1,59 @@
-rebase.
-
-Once you are done fixing conflicts for a specific commit, run:
-
-```bash
-git rebase --continue
-```
-
-This will continue the rebasing process. Once you are done fixing all
-conflicts you should run the existing tests to make sure you didn’t break
-anything, then run your new tests (there are new tests, right?) and
-make sure they work also.
-
-If rebasing broke anything, fix it, then repeat the above process until
-you get here again and nothing is broken and all the tests pass.
-
-### Make a pull request
-
-Make a clear pull request from your fork and branch to the upstream master
-branch, detailing exactly what changes you made and what feature this
-should add. The clearer your pull request is the faster you can get
-your changes incorporated into this repo.
-
-At least one other person MUST give your changes a code review, and once
-they are satisfied they will merge your changes into upstream. Alternatively,
-they may have some requested changes. You should make more commits to your
-branch to fix these, then follow this process again from rebasing onwards.
-
-Once you get back here, make a comment requesting further review and
-someone will look at your code again. If they like it, it will get merged,
-else, just repeat again.
-
-Thanks for contributing!
-
-### Guidelines
-
-1. Uphold the current code standard:
-    - Keep your code [DRY][].
-    - Apply the [boy scout rule][].
-    - Follow [STYLE-GUIDE.md](STYLE-GUIDE.md)
-1. Run the [tests][] before submitting a pull request.
-1. Tests are very, very important. Submit tests if your pull request contains
-   new, testable behavior.
-1. Your pull request is comprised of a single ([squashed][]) commit.
+  - `git commit -m "(fix) Fixed inconsistent tests [Fixes #0]"`
+  - Keywords for closing issues - The following keywords will close an issue via commit message:  
+    +close  
+    +closes  
+    +closed  
+    +fix  
+    +fixes  
+    +fixed  
+    +resolve  
+    +resolves  
+    +resolved  
+5. Make changes to your feature branch. When ready to make a pull request, pull down changes from upstream master to feature branch. 
+  -git pull --rebase upstream master
+  -Because rebase changes commits, you will have to push with the -f or --force flag to your branch after rebasing, as the history has changed in a way that git cannot resolve. You should never rebase or push with force to the master branch of your repository, as that will invalidate everyone elses' clones and checkouts of the repository.
+   directly to master. Include a description of your changes.
+6. If you run into merge conflict issues, correct them and then continue the rebase. You pick a file by git adding it. Do not make commits during a rebase. Once you are done fixing conflicts for a specific commit, run:
+  `git rebase --continue`
+7. Push to your fork's feature branch  **** NEVER PUSH ORIGIN UPSTREAM MASTER *** WE ALL HAVE RIGHTS TO DO IT, BUT IT CAN COMPROMISE OUR CODE
```

Comment:
We can probably take this ***NEVER push section out as it applied to our specific case where we each had author rights.